stdlib/public/core/OutputStream.swift

```diff
@@ -330,9 +330,19 @@ internal struct _Stdout : OutputStreamType {
   }
 
   mutating func write(string: String) {
-    // FIXME: buffering?
-    // It is important that we use stdio routines in order to correctly
-    // interoperate with stdio buffering.
+    if string.isEmpty { return }
+
+    if string._core.isASCII {
+      defer { _fixLifetime(string) }
+
+      let result = _swift_stdlib_fwrite_stdout(
+        UnsafePointer(string._core.startASCII), string._core.count, 1)
+      if result != 1 {
+        fatalError("fwrite() returned \(result), expected 1")
```

Comment:
And aborting the program over a mere failure to print to stdout seems rather extreme. If we can't print to stdout, presumably that means there's nothing listening on the other end of the pipe anymore, so it's not like there's anything that would have been able to see our output if we did successfully print. The only other obvious reason for a print failure is if someone wrote something like `fclose(stdout)`, which means they intentionally shut off the ability to print.